src/ImageSearchModal.jsx

```diff
@@ -15,238 +15,203 @@ import cockpit from 'cockpit';
-                       <Form isHorizontal className="image-search-tag-form">
-                           <FormGroup fieldId="image-search-tag" label={_("Tag")}>
-                               <TextInput className="image-tag-entry"
-                                      id="image-search-tag"
-                                      type='text'
-                                      placeholder="latest"
-                                      value={this.state.imageTag || ''}
-                                      onChange={(_event, value) => this.onValueChanged('imageTag', value)} />
-                           </FormGroup>
-                       </Form>
-                       <Button variant='primary' isDisabled={this.state.selected == undefined} onClick={this.onDownloadClicked}>
-                           {_("Download")}
-                       </Button>
-                       <Button variant='link' className='btn-cancel' onClick={Dialogs.close}>
-                           {_("Cancel")}
-                       </Button>
-                   </>}
-            >
-                {defaultBody}
-            </Modal>
-        );
-    }
-}
+    };
+
+    const onToggleUser = ev => setIsSystem(ev.currentTarget.value === "system");
+    const onDownloadClicked = () => {
+        const selectedImageName = imageList[selected].Name;
+        if (activeConnection)
+            activeConnection.close();
+        Dialogs.close();
+        downloadImage(selectedImageName, imageTag, isSystem);
+    };
+
+    const handleClose = () => {
+        if (activeConnection)
+            activeConnection.close();
+        Dialogs.close();
+    };
+
+    return (
+        <Modal isOpen className="podman-search"
+               position="top" variant="large"
+               onClose={handleClose}
+               title={_("Search for an image")}
+               footer={<>
+                   <Form isHorizontal className="image-search-tag-form">
+                       <FormGroup fieldId="image-search-tag" label={_("Tag")}>
+                           <TextInput className="image-tag-entry"
+                                  id="image-search-tag"
+                                  type='text'
+                                  placeholder="latest"
+                                  value={imageTag || ''}
+                                  onChange={(_event, value) => setImageTag(value)} />
+                       </FormGroup>
+                   </Form>
+                   <Button variant='primary' isDisabled={selected === ""} onClick={onDownloadClicked}>
+                       {_("Download")}
+                   </Button>
+                   <Button variant='link' className='btn-cancel' onClick={handleClose}>
+                       {_("Cancel")}
+                   </Button>
+               </>}
+        >
+            <Form isHorizontal>
+                {dialogError && <ErrorNotification errorMessage={dialogError} errorDetail={dialogErrorDetail} />}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230705-092041-36f39867-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)